bowling/file.cpp

```diff
@@ -1,23 +1,86 @@
 #include "file.hpp"
 
-#include <experimental/filesystem>
+#include <algorithm>
+#include <filesystem>
 #include <fstream>
 #include <iostream>
+#include <iterator>
+#include <sstream>
 
-File::File(std::string fileName)
-    : fileName_(fileName) {}
+File::File() : fileName_("../lane.txt"){};
+
+File::File(std::string fileName) : fileName_(fileName) {}
+
+std::vector<std::pair<int, int>> File::getScores() const {
+    return score_;
+}
+
+std::string File::getResults() const {
+    return results_;
+}
+
+void File::splitString(const std::string& str, std::vector<std::string>& vec) {
+    std::stringstream ss(str);
+    std::string token;
+    while (std::getline(ss, token, delimiter)) {
+        vec.push_back(token);
+    }
+}
+
+void File::translateResultsToScoreVector(std::string results) {
+    std::vector<std::string> results_divided;
+    std::vector<int> temp{};
+
+    this->splitString(results, results_divided);
+
+    for (auto elem : results_divided) {
+        if (elem.empty()) {
+            continue;
+        } else if (elem == "X") {
+            this->score_.push_back(std::make_pair(10, 0));
+        } else if (elem == "XX") {
+            this->score_.push_back(std::make_pair(10, 0));
+            this->score_.push_back(std::make_pair(10, 0));
+        } else if (elem == "--") {
+            this->score_.push_back(std::make_pair(0, 0));
+        } else {
+            for (int i = 0; i < 2; i++) {
+                if (elem[i] == '-') {
+                    temp.push_back(0);
+                } else if (std::isdigit(elem[i])) {
+                    temp.push_back((int)elem[i] - 48);

```

Comment:
You can use std::atoi(&elem[i])

Reply:
Unfortunately I checked this solution and it was not working :/ I will try to change it a bit, but it will be hard ;)